DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs

```diff
@@ -2141,4 +2150,189 @@ internal class ContextMenuCommand
         /// </summary>
         public MethodInfo Method { get; set; }
     }
+
+    #region Default Help
+    /// <summary>
+    /// Represents the default help module.
+    /// </summary>
+    public class DefaultHelpModule : ApplicationCommandsModule
+    {
+        public class DefaultHelpAutoCompleteProvider : IAutocompleteProvider
+        {
+            public async Task<IEnumerable<DiscordApplicationCommandAutocompleteChoice>> Provider(AutocompleteContext context)
+            {
+                var options = new List<DiscordApplicationCommandAutocompleteChoice>();
+
+                var globalCommands = await context.Client.GetGlobalApplicationCommandsAsync();
+                var guildCommands = await context.Client.GetGuildApplicationCommandsAsync(context.Guild.Id);
+                var slashCommands = new List<DiscordApplicationCommand>();
```

Comment:
I would suggest this alternative. Where you invoke the Distinct method and convert it to a list -- it doesn't update `slashCommands` as you might think.

Also might want to specify what makes it `distinct` which I assume in this context is the name.

```csharp
var slashCommands = globalCommands.Concat(guildCommands)
     .Where(ac => !ac.Name.Equals("help", StringComparison.OrdinalIgnoreCase))
     .DistinctBy(ac => ac.Name)
     .ToList();
```

Reply:
Doh missed that! Yes you're right will get that changed, think there may be some other cases where i've done the same so will get those done too

Reply:
Hmmm unless i'm missing something on this suggestion, `.DistinctBy()` only exists inside the MoreLinq package and therefore wouldn't be possible here.

Reply:
Odd, it appears for me under System.Linq

Reply:
I can see it on Microsoft website in docs, released in .Net 6, but its not recognised in either my VS or Rider both set at .Net 6 so if I add it won't compile.
Have reinstalled my .Net and verified i'm using .Net 6 so no reason it shouldn't show but can't for the life of me get it to show

Reply:
I now understand what you meant, I didn't realize the library was targeting netstandard 2.0 which is why DistinctBy isn't there. When testing the changes I suggested I picked one of the unit tests which is using the later versions... hence it existed. That's my b.

@Lulalaby at what point are we going to up the framework version? Microsoft is recommending Net5+. Though I'm not sure how many developers of the community are using pre Net5

Reply:
That's a fair point to discuss with everyone on the guild in the new year.
For now I feel that the safest is to go with 2.1

Reply:
The modules I did, hosting and configuration are net6... I kinda... didn't realize... thatttttt we were on netstandard